app/src/hooks/useDownload.tsx

```diff
@@ -0,0 +1,53 @@
+import { useDialogContext } from './useContext';
+
+const useDownload = () => {
+  const dialogContext = useDialogContext();
+  /**
+   * handler for downloading raw data as JSON
+   * Note: currently this does not zip the file. Can be modified if needed.
+   *
+   * @param {any} data - to download
+   * @param {string} fileName - name of file excluding file extension ie: file1
+   */
+  const downloadJSON = (data: any, fileName: string) => {
+    const blob = new Blob([JSON.stringify(data, undefined, 2)], { type: 'application/json' });
+
+    const link = document.createElement('a');
+
+    link.download = `${fileName}.json`;
```

Comment:
It probably makes sense to sanitize the filename. For example, it shouldn't allow any special characters or trailing whitespace, and should probably have a max length. Could be its own util method.

Reply:
Yea this is a good idea. I'll implement this.